### DISCUSSION TITLE
Questions about my Z-Wave network speed

### DISCUSSION THREAD

**Discussion by alexojegu** [zwave-js/zwave-js-ui#3855] [Q&A] [answered]
I have 1 controller (Aeotec Z-Stick 7) and 11 devices (Shelly Qubino Wave Shutter) and this is my network graph:
![image](https://github.com/user-attachments/assets/b5ffd91e-f17d-4fa0-832a-7ff8ebd37157)

Devices from "Persiana 6" to "Persiana 11" are all on same room that the controller and all them to less from 10 meters of this. So I understand the controller can only transmit to one speed (The maximum speed of the controller is the speed of the slowest directly connected node), is this true?

Is it better have the devices direct connect to the controller to 40 kbit/s than have hops but 100 kbit/s?

To customize the network routing to try to get 100 kbit/s with hops, Do I have to define `Priority route` or `Return routes` or both?

Some other relevant data:

![image](https://github.com/user-attachments/assets/2405a370-14cd-48a6-bf45-2d361695575b)

![image](https://github.com/user-attachments/assets/69efe8f8-50d1-4a95-8812-f22dc0b5aaf5)
![image](https://github.com/user-attachments/assets/5a46d50c-0e9a-4625-aad6-1c1f18cc272e)

The controller is configured as Aeotec recommended:

> For EU Frequency:
> - TX Power Level Normal = 9.9 dBm (or 99 value)
> - TX Power Level Measured = -2.3 dBm (or -23 value)

**Comment by robertsLando** [comment]
Sorry for late reply but I was on vacation.

The best that can anwser to those questions is @AlCalzone , anyway if you don't have communications problems I would not worry about this

**Comment by AlCalzone** [comment, marked as answer]
> So I understand the controller can only transmit to one speed (The maximum speed of the controller is the speed of the slowest directly connected node), is this true?

No. The controller stores routes and speed for each node separately.

I'm not sure why you're only getting 40 kbps when all devices support 100, but the controller placement may have something to do with it. I suggest you check out https://zwave-js.github.io/node-zwave-js/#/troubleshooting/first-steps?id=first-steps for some troubleshooting steps.
Side note: I haven't had a good experience with the Z-Stick 7 in general

**Reply by alexojegu** [reply]
Sorry for my late response, little time.

I was trying to move my Z-Wave JS UI from router (Turris Omnia -> OpenWRT -> LXC (Aline Linux) -> Z-Wave JS) to my old ARMv7 SBC (Wandboard -> Arch Linux ARM -> Docker / Podman -> Z-Wave JS UI) but the system hangs.

It seems to be related to https://github.com/zwave-js/zwave-js-ui/issues/3639 and https://github.com/raspberrypi/linux/issues/6172 but I am using Arch Linux ARM with Kernel 6.10.7, maybe a main line Kernel bug.

@AlCalzone, if AEOTEC is collaborative, would it be possible to report it? I don't have enough resources to do a Git Bisect and report directly to the main line Kernel.

**Reply by AlCalzone** [reply]
I'm not sure there's much they can do about the Linux kernel issues. AFAIK all 700+ series sticks (and a lot of other serial peripherals) use a Silicon Labs CP2102N to provide communication with the host.

**Reply by alexojegu** [reply]
I understand that they will have more resources and are the main interested party in getting their device working. So if they can determine the exact version of the first affected Kernel and do a Git Bisect between this version and the previous one, it is possible to report it to Kernel Bugzilla and they will surely fix it quickly (without the Git Bitsect the bug will probably remain open indefinitely).

I've done it before but on x86_64 architecture, doing it on ARMv7 and without Distcc Cross-Compiling... is a long, long time. (Build a Kernel on my Intel i7 6th gen with 16G RAM takes ~1h, build it on my ARMv7 with 2G RAM can take ~8h or probably more).

Or maybe they can report it to Silicon Labs, AEOTEC does not seem to be the only one affected: https://github.com/zwave-js/zwave-js-ui/issues/3639#issuecomment-2308620769

**Reply by alexojegu** [reply]
This weekend I saw some errors on the disk so I changed the SSD and reinstalled it updating the kernel to version 6.10.10. It has been working fine for about 24 hours. Maybe the problem was the SSD, it's strange because it only rebooted when starting Z-Wave JS UI with only Home Assistant started it worked fine. At any case sorry for the noise @AlCalzone.

About the main topic of discussion, the network graph now looks fine:
![image](https://github.com/user-attachments/assets/fe9b9b66-9aa5-4e50-b4a9-37d60076c4cc)
